src/solvers/smt2/smt2_conv.cpp

```diff
@@ -283,6 +283,7 @@ exprt smt2_convt::get(const exprt &expr) const
 
     if(it!=identifier_map.end())
       return it->second.value;
+    return expr;
   }
```

Comment:
It's wrong for `get` to return the symbol that it's asked to provide a model for.
The API is meant to return `nil` when no value is available.

Reply:
This appears to have been caught from another PR/merging problem with git (and is already merged in #6210 ). That said, this fixes missing symbols in arrays when printing traces from SMT solvers.